packages/store/src/configs/messages.config.ts

```diff
@@ -30,15 +17,53 @@ export const CONFIG_MESSAGES = {
     'NgxsModule.forRoot(states, { developmentMode: !environment.production })',
   [VALIDATION_CODE.SELECT_FACTORY_NOT_CONNECTED]: () =>
     'You have forgotten to import the NGXS module!',
-  [VALIDATION_CODE.ACTION_DECORATOR]: () =>
-    '@Action() decorator cannot be used with static methods',
-  [VALIDATION_CODE.SELECTOR_DECORATOR]: () => 'Selectors only work on methods',
-  [VALIDATION_CODE.ZONE_WARNING]: () =>
-    'Your application was bootstrapped with nooped zone and your execution strategy requires an actual NgZone!\n' +
-    'Please set the value of the executionStrategy property to NoopNgxsExecutionStrategy.\n' +
-    'NgxsModule.forRoot(states, { executionStrategy: NoopNgxsExecutionStrategy })',
+  // This can be a breaking change if we don't throw these errors even in production mode.
   [VALIDATION_CODE.PATCHING_ARRAY]: () => 'Patching arrays is not supported.',
-  [VALIDATION_CODE.PATCHING_PRIMITIVE]: () => 'Patching primitives is not supported.',
-  [VALIDATION_CODE.UNDECORATED_STATE_IN_IVY]: (name: string) =>
-    `'${name}' class should be decorated with @Injectable() right after the @State() decorator`
+  [VALIDATION_CODE.PATCHING_PRIMITIVE]: () => 'Patching primitives is not supported.'
 };
+
+// The below functions are decoupled from the `CONFIG_MESSAGES` object for now, since object properties
+// are not tree-shakable. That means that if the error is thrown only in development mode it still will get
+// bundled into the final file. This is how Angular does error tree-shaking internally.
```

Comment:
I really don't like the enum anyway. I feel like it is overengineering and the function approach below is simpler and cleaner. Also, the enum is not an `as const` so it results in a large amount of js code generated.

Reply:
`I really don't like the enum anyway.` we can fix it later)

Reply:
`I really don't like the enum anyway.` we can fix it later)